test/fec_utest.cpp

```diff
@@ -66,21 +76,27 @@ class FecTestCommon : public ::testing::Test {
                     props[i] = quadiron::Properties();
                 }
             }
-            for (int i = 0; i < n_data; i++) {
-                v.set(i, gf->weak_rand());
+
+            // gen_data(gf, data_frags, is_nf4);
+            for (unsigned i = 0; i < n_data; i++) {
+                data_frags.set(i, wk_rand());
```

Comment:
Since:
- we are in test, perf is not critical (anyway, using std::function is not for perf critical code :P)
- `wk_rand` is used only here

Why not simplifying the code as:

```c++
for (unsigned i = 0; i < n_data; i++) {
    const T v = is_nf4 ? nf4.unpacked_weak_rand() : gf.weak_rand();
    data_frags.set(i, v);
}
```

?